services/statestorage/service.go

```diff
@@ -16,31 +17,49 @@ import (
 
 var LogTag = log.Service("state-storage")
 
+type stateIncrement struct {
+	diff       adapter.ChainDiff
+	merkleRoot primitives.MerkleSha256
+	height     primitives.BlockHeight
+	ts         primitives.TimestampNano
+}
+
 type service struct {
 	config       config.StateStorageConfig
-	merkle       *merkle.Forest
 	blockTracker *synchronization.BlockTracker
 	logger       log.BasicLogger
 
-	mutex                    *sync.RWMutex
-	persistence              adapter.StatePersistence
-	lastCommittedBlockHeader *protocol.ResultsBlockHeader
+	mutex       sync.RWMutex
```

Comment:
i would move the state incement, the incache and the height and ts to a seperate file with all the _methods so that it will be in charge of dealing with the history + persistent ....

question as merkle needs to be synced ... do we also have that there ?